test/LEAF_Request_Portal_Tests/tests/api/WorkflowControllerTest.php

```diff
@@ -0,0 +1,205 @@
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+    }
+
+    /**
+     * Tests the POST `workflow/dependency/[digit]` endpoint.
+     */
+    public function testUpdateDependency() : void
+    {
+      $data = array(
+        'description' => 'Updated Dependency Title'
+      );
+      $response = self::$client->postEncodedForm('?a=workflow/dependency/8', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+    }
+
+    /**
+     * Tests the POST `workflow/dependency/[digit]/privileges` endpoint.
+     */
+    public function testGrantDependencyPrivs() : void
+    {
+      $data = array(
+        'groupID' => 3
+      );
+
+      $response = self::$client->postEncodedForm('?a=workflow/dependency/8/privileges', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+    }
+
+    /**
+     * Tests the POST `workflow/[digit]/step/[digit]/[text]/events` endpoint.
+     */
+    public function testLinkEvent() : void
+    {
+      $data = array(
+        'eventID' => 'std_email_notify_completed'
+      );
+
+      $response = self::$client->postEncodedForm('?a=workflow/1/step/1/_approve/events', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+    }
+
+
+    /**
+     * Tests the POST `workflow/dependencies` endpoint.
+     */
+    public function testAddDependency() : void
+    {
+      $data = array(
+        'description' => 'Test Dependency'
+      );
+      $response = self::$client->postEncodedForm('?a=workflow/dependencies', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(9, $response);
+    }
+
+
+    /**
+     * Tests the DELETE `workflow/step/[digit]/dependencies` endpoint.
+     */
+    public function testUnlinkDependency() : void
+    {
+      $data = array(
+        'dependencyID' => 8
+      );
+      $response = self::$client->postEncodedForm('?a=workflow/step/1/dependencies', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+
+      $delResponse = self::$client->Delete('?a=workflow/step/1/dependencies&dependencyID=8');
+      $this->assertNotNull($delResponse);
+      $this->assertEquals(true, $response);
+    }
+
+
+    /**
+     * Tests the DELETE `workflow/[digit]/step/[digit]/[text]/events` endpoint.
+     */
+    public function testUnlinkEvent() : void
+    {
+      $data = array(
+        'eventID' => 'std_email_notify_completed'
+      );
+
+      $response = self::$client->postEncodedForm('?a=workflow/1/step/1/_approve/events', $data);
+      $this->assertNotNull($response);
+      $this->assertEquals(true, $response);
+
+      $event_id = self::$client->get('?a=workflow/1/step/1/_approve/events');
+
+      $delResponse = self::$client->Delete('?a=workflow/1/step/1/_approve/events&eventID=' . $event_id['eventID']);
```

Comment:
The `LEAFClient::delete()` method is not present in this branch, so this test is failing.

Reply:
done

Reply:
This test is failing. `$event_id` is an array, so an index check needs to be done, and to access the `eventID` you need to get that index (`$event_id[0]['eventID']`).